test/Microsoft.AspNet.Routing.Tests/DefaultInlineConstraintResolverTest.cs

```diff
@@ -287,6 +298,21 @@ public void ResolveConstraint_AmbiguousConstructors_Throws()
                          ex.Message);
         }
 
+        // These are cases which parsing does not catch and we'll end up here
+        [Theory]
+        [InlineData("regex(abc")]
+        [InlineData("int/")]
+        [InlineData("in{t")]
+        public void ResolveConstraint_Invalid_Throws(string constraint)
+        {
+            // Arrange
+            var routeOptions = new RouteOptions();
+            var resolver = GetInlineConstraintResolver(routeOptions);
+
+            // Act & Assert
+            Assert.Null(resolver.ResolveConstraint(constraint));
```

Comment:
I want to see an integration test like this, including the exception that's thrown. It should look like this

`new TemplateRoute("{c}/{a}/{ssn:regex(foo}");`